pkg/api/folder.go

```diff
@@ -648,17 +649,43 @@ func newFolderK8sHandler(hs *HTTPServer) *folderK8sHandler {
 	}
 }
 
+func (fk8s *folderK8sHandler) searchFolders(c *contextmodel.ReqContext) {
+	client, ok := fk8s.getClient(c)
+	if !ok {
+		return // error is already sent
+	}
+	out, err := client.List(c.Req.Context(), v1.ListOptions{})
+	if err != nil {
+		fk8s.writeError(c, err)
+		return
+	}
+
+	query := strings.ToUpper(c.Query("query"))
+	folders := []folder.Folder{}
+	for _, item := range out.Items {
+		p := internalfolders.UnstructuredToLegacyFolder(item)
+		if p == nil {
+			continue
+		}
+		if query != "" && !strings.Contains(strings.ToUpper(p.Title), query) {
+			continue // query filter
+		}
+		folders = append(folders, *p)
+	}
+	c.JSON(http.StatusOK, folders)
+}
+
 func (fk8s *folderK8sHandler) createFolder(c *contextmodel.ReqContext) {
 	client, ok := fk8s.getClient(c)
 	if !ok {
 		return // error is already sent
 	}
-	cmd := folder.CreateFolderCommand{}
+	cmd := folder.UpdateFolderCommand{}
```

Comment:
Do we need to replace this? I saw that Playlists use `UpdatePlaylistCommand` but I am not sure that we do.

Reply:
Fixed